apps/champions/lib/champions/battle/simulator.ex

```diff
@@ -516,20 +511,47 @@ defmodule Champions.Battle.Simulator do
 
   # Choose the targets for an effect with "random" as the strategy. Returns the target ids.
   # The `== target_allies` works as a negation operation when `target_allies` is `false`, and does nothing when `true`.
-  defp choose_targets(
-         %{team: team} = _caster,
-         %{
-           count: count,
-           type: "random",
-           target_allies: target_allies
-         },
-         state
-       ),
-       do:
-         state.units
-         |> Enum.filter(fn {_id, unit} -> unit.team == team == target_allies end)
-         |> Enum.take_random(count)
-         |> Enum.map(fn {id, _unit} -> id end)
+  defp choose_targets(caster, %{count: count, type: "random", target_allies: target_allies}, state),
+    do:
+      state.units
+      |> Enum.filter(fn {_id, unit} -> unit.team == caster.team == target_allies end)
+      |> Enum.take_random(count)
+      |> Enum.map(fn {id, _unit} -> id end)
+
+  defp choose_targets(caster, %{count: count, type: "nearest", target_allies: target_allies}, state) do
+    config_name = if target_allies, do: :ally_proximities, else: :enemy_proximities
+
+    state.units
+    |> Enum.map(fn {_id, unit} -> unit end)
+    |> Enum.filter(fn unit -> unit.team == caster.team == target_allies and unit.id != caster.id end)
+    |> find_by_proximity(
+      Application.get_env(:champions, :"slot_#{caster.slot}_proximities")[config_name],
+      count
+    )
+    |> Enum.map(& &1.id)
```

Comment:
I tested using the commands that you provided in the PR description, and I found that both nearest and furthest cases result in more than one unit targeted. 
This is what I got in the log after inspecting the Enum.map/2 result:

[info] Process step 58 for unit Melo-1e7
[info] Process step 58 for unit Melo-4ac
[info] Process step 58 for unit Muflus-937
nearest: ["1e7302e1-656b-4407-ac70-af43b960217d", "4ac005f9-4612-44e5-926d-cf81f4f46a78"]
(...)
[info] Process step 65 for unit Melo-1e7
[info] Process step 65 for unit Melo-4ac
[info] Process step 65 for unit Muflus-937
[info] Unit Muflus-937 cannot attack because it is casting another skill
[info] Process step 65 for skill Muflus Basic cast by 937
[info] Trigger delay for mechanic eed of skill 820abe00-4f27-4b0d-8204-0070f6316bbc ready.
[info] Muflus-937's effect is ready to be processed
[info] Dealing 34 damage to Melo-1e7 (254 -> 220)
[info] Dealing 34 damage to Melo-4ac (254 -> 220)
[info] Step 65 finished: %{units: %{"Team 1" => [%{unit: "937", cooldown: 55, energy: 100, health: 621}], "Team 2" => [%{unit: "1e7", cooldown: 42949606, energy: 50, health: 220}, %{unit: "4ac", cooldown: 42949606, energy: 50, health: 220}]}, skills_being_cast: [%{name: "Muflus Basic", animation_duration: 4, caster_id: "9373c324-85a9-40fe-a995-182944b01f3a"}], pending_effects: []}